src/connection.cpp

```diff
@@ -148,6 +157,35 @@ void Connection::parseHeader(const boost::system::error_code& error)
 		return;
 	}
 
+	if (!receivedLastChar && connectionState == CONNECTION_STATE_CONNECTING_STAGE2) {
+		uint8_t* msgBuffer = msg.getBuffer();
+
+		if (!receivedName && msgBuffer[1] == 0x00) {
+			receivedLastChar = true;
+		} else {
+			std::string serverName = g_config.getString(ConfigManager::SERVER_NAME) + "\n";
```

Comment:
this is not used?

Reply:
Leftover from removing a check for matching server name.